src/main/markers/waveform.segments.js

```diff
@@ -25,10 +25,10 @@ define([
       return view;
     });
 
+    var segmentId = 0;
     var createSegmentWaveform = function (options) {
-      var segmentId = "segment" + self.segments.length;
       var segment = {
-        id: segmentId,
+        id: options.id || ('peaks.segment.' + segmentId++),
```

Comment:
I see the `set_point` method in waveform-data handles this correctly, but `set_segment` is inconsistent. I'll fix and update it.